Xcode compile error in Backtrace.cc, after #140 merged

PR #140 introduces an implicit conversion from `size_t` to `unsigned`, which causes the Xcode build to break (for macOS, probably also iOS):

```
fleece/Fleece/Support/Backtrace.cc:141:35: error: implicit conversion loses integer precision: 'decltype(this->_addrs.size())' (aka 'unsigned long') to 'unsigned int' [-Werror,-Wshorten-64-to-32]
            auto frame = getFrame(i);
                         ~~~~~~~~ ^
```

~~I don't know why this wasn't caught by CI; the macOS build there passed.~~ It passed CI because CMake doesn't enable `-Wshorten-64-to-32` Just ran a CMake build on my Mac on the same commit and it passed. 

I distinctly remember setting up an `xcodebuild` task in Github Actions, which would have caught this; hmm.

--- a/Fleece/Support/Backtrace.cc
+++ b/Fleece/Support/Backtrace.cc
@@ -138,7 +138,7 @@ namespace fleece {
                 out << '\n';
             out << '\t';
             char *cstr = nullptr;
-            auto frame = getFrame(i);
+            auto frame = getFrame(unsigned(i));
             int len;
             bool stop = false;
             if (frame.function) {

